### PR TITLE
Added test actions

### PR DESCRIPTION
**Pull request by leshikus**
1) added clang test;
2) added test action;
3) added Ilia's patch for OOM in ResultSetTest;
4) added 8 Gb memory to the runner and changed memory manager settings

### CODE REVIEW

File: Tests/ResultSetTest.cpp [hunk 1]
```diff
@@ -1870,37 +1870,59 @@ TEST(Reduce, BaselineHashColumnar) {
 #ifndef HAVE_TSAN
 // The large buffers tests allocate too much memory to instrument under TSAN
 TEST(ReduceLargeBuffers, PerfectHashOne_Overflow32) {
-  const auto target_infos = generate_random_groups_nullable_target_infos();
-  auto query_mem_desc = perfect_hash_one_col_desc(target_infos, 8, 0, 222208903, {8});
-  EvenNumberGenerator gen1;
-  EvenNumberGenerator gen2;
-  test_reduce(target_infos, query_mem_desc, gen1, gen2, 2, false);
+  try {
```
Comment (Garra1980) [inline Tests/ResultSetTest.cpp:1873]: let's probably change this in separate PR? cc @ienkovich

Comment (Garra1980) [reply Tests/ResultSetTest.cpp:1873]: we may probably already have #384

**Review by ienkovich — COMMENTED** [review]
The clang-format check shows that all or almost all files fail to pass the check. But most developers have clang-format enabled on their SDK and I believe that most of the files are formatted correctly. We should check that a proper version of clang-format is used and .clang-format file is respected in the check.

**Comment by leshikus** [comment]
Rebased, updated the pull request with the clang fix

**Pull request closed without merging**